cmd/soroban-rpc/internal/db/transaction.go

```diff
@@ -135,48 +139,36 @@ func (txn *transactionHandler) trimTransactions(latestLedgerSeq uint32, retentio
 	return err
 }
 
-// GetLedgerRange pulls the min/max ledger sequence numbers from the database.
+// GetLedgerRange pulls the min/max ledger sequence numbers from the transactions table.
 func (txn *transactionHandler) GetLedgerRange(ctx context.Context) (ledgerbucketwindow.LedgerRange, error) {
 	var ledgerRange ledgerbucketwindow.LedgerRange
 
-	//
-	// We use subqueries alongside a UNION ALL stitch in order to select the min
-	// and max from the ledger table in a single query and get around sqlite's
-	// limitations with parentheses (see https://stackoverflow.com/a/22609948).
-	//
-	newestQ := sq.
-		Select("m1.meta").
-		FromSelect(
-			sq.
-				Select("meta").
-				From(ledgerCloseMetaTableName).
-				OrderBy("sequence ASC").
-				Limit(1),
-			"m1",
-		)
-	sql, args, err := sq.
-		Select("m2.meta").
-		FromSelect(
-			sq.
-				Select("meta").
-				From(ledgerCloseMetaTableName).
-				OrderBy("sequence DESC").
-				Limit(1),
-			"m2",
-		).ToSql()
-	if err != nil {
-		return ledgerRange, fmt.Errorf("couldn't build ledger range query: %w", err)
+	// Query to get the minimum and maximum ledger sequence from the transactions table
+	var ledgerSeqs struct {
+		MinLedgerSequence *uint32 `db:"min_ledger_sequence"`
+		MaxLedgerSequence *uint32 `db:"max_ledger_sequence"`
+	}
+	minMaxLedgerSequenceSQL := sq.
+		Select("MIN(ledger_sequence) AS min_ledger_sequence, MAX(ledger_sequence) AS max_ledger_sequence").
+		From(transactionTableName)
+	if err := txn.db.Get(ctx, &ledgerSeqs, minMaxLedgerSequenceSQL); err != nil {
+		return ledgerRange, fmt.Errorf("couldn't query ledger range: %w", err)
+	}
+
+	// Empty DB or just a single row in the DB.
+	if ledgerSeqs.MinLedgerSequence == nil || ledgerSeqs.MaxLedgerSequence == nil {
+		return ledgerRange, nil
 	}
 
+	// Use the min and max ledger sequences to query the ledger_close_meta table
+	ledgerMetaSQL := sq.
+		Select("lcm.meta").
+		From(ledgerCloseMetaTableName + " as lcm").
+		Where(sq.Eq{"lcm.sequence": []uint32{*ledgerSeqs.MinLedgerSequence, *ledgerSeqs.MaxLedgerSequence}})
+
```

Comment:
You should be able to do it in a single query to the DB

Reply:
Yeah I did not realise combining everything in single would be so efficient.